src/fondant/pipeline/lightweight_component.py

```diff
@@ -121,6 +172,32 @@ class PythonComponentOp(cls, PythonComponent):
             def image(cls) -> Image:
                 return image
 
+            @classmethod
+            def consumes(cls) -> t.Optional[list]:
+                return consumes
+
+            @classmethod
+            def get_consumes_spec(
```

Comment:
Can we lift this to the base class?

I would try to keep the decorator as simple as possible. Only what is needed at decoration time should be inside it. Everything else I would try to move to a logical place outside of the decorator.